### ISSUE TITLE
🐛 BUG: CSS HMR causes flashes

### ISSUE THREAD

**Issue by AllanChain** [withastro/astro#3370]
### What version of `astro` are you using?

1.0.0-beta.28

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

When updating CSS in `.astro` file, the HMR is not working properly and causes the page to flash.

#### Repro

In the repro link, modify the color, and you will see the title first flashes back to black and aligns to the left side, then the correct style is applied.

https://user-images.githubusercontent.com/36528777/168453843-b9c5cdc6-17e8-457d-adde-750626443506.mp4

#### Additional info

Possible related to the problem that the CSS is loading twice in each HMR.

<img width="594" alt="devtool-screenshot" src="https://user-images.githubusercontent.com/36528777/168453451-1a485760-fb62-4f33-abe5-4df3ddaa18b6.png">

The first one is loaded by vite and the second one is triggered by html replacement.

Also possibly related to #3219 

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/withastro-astro-mtxvii?file=src/pages/index.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
@natemoo-re I looked into this and I don't believe it's a regression. Here's the same thing happening in .27: https://stackblitz.com/edit/withastro-astro-liqoye?file=src%2Fpages%2Findex.astro

What's happen is that Vite's HMR appends `?t=TIMESTAMP` to the links, and non-CSS is returned, so we only get correct styles after the micromorph patching completes. This results in the flash.

**Comment by AllanChain** [comment]
@matthewp I don't think this issue was solved in beta.38. I have updated the Astro version in the Stackblitz example. Could you check it again?
